hca_util/__main__.py

```diff
@@ -93,5 +96,19 @@ def parse_args(args):
 
 
 if __name__ == '__main__':
-    parsed_args = parse_args(sys.argv[1:])
-    HcaCmd(parsed_args)
+    try:
+        parsed_args = parse_args(sys.argv[1:])
+        HcaCmd(parsed_args)
+    except KeyboardInterrupt:
+        # If SIGINT is triggered whilst threads are active (upload/download) we kill the entire process to give the
+        # user an instant exist, rather than have to hammer on ctrl+c multiple times with various obscure messages.
+        #
+        # However, os_.exit() is nasty because it allows the Python interpreter to do no cleanup
+        # One alternative is to make our transfer threads daemon. However, there are then other non-daemon threads
+        # employed by boto for the multi-part transfer. Here, configuring boto to only use the main thread for transfer
+        # works but with unmeasured effects on transfer speed. Alternatively, it might be possible to call
+        # boto3 client.abort_multipart_upload() but don't know for sure and in any case this requires more work to
+        # keep hold of upload IDs, etc.
+        #
+        # So for now risking the kill though might want to revisit this decision.
+        os._exit(0)
```

Comment:
i'd use the `signal` module to handle this in a neater way, incl. as you said in the comment making sure multipart upload is aborted correctly.

Reply:
[You can't send signals to threads other than main](https://docs.python.org/3/library/signal.html#signals-and-threads). AFAICS, the most we could do is see if calling `client.abort_multipart_upload()` will properly abort boto3 threads but this is not guaranteed (the doc doesn't say you can do this). It will also require a non-trivial amount of engineering so right now I propose to keep using `os._exit(0)`. Yeah, I don't like it much either.